### PR TITLE
fix(appkit): add jitter to RetryInterceptor exponential backoff

### PR DESCRIPTION
**Pull request by MarioCadenas**
## Summary
The `RetryInterceptor` used pure exponential backoff (`initialDelay * 2^attempt`), causing all concurrent retries after a simultaneous failure to fire at identical intervals — a thundering herd problem. This PR adds full jitter by default: each retry delay is randomized between 50% and 100% of the computed backoff value, spreading retries over time.

## Changes
- Added `jitter?: boolean` field to `RetryConfig` (defaults to `true`)
- Updated `RetryInterceptor.calculateDelay()` to apply random jitter: `capped * (0.5 + Math.random() * 0.5)`
- Existing tests updated to set `jitter: false` where exact timing is asserted
- New tests: jitter enabled by default, boundary behavior at `Math.random() = 0` (50%) and `Math.random() = 1` (100%)

## Test plan
- [x] Existing tests pass (backoff + maxDelay tests now use `jitter: false`)
- [x] New tests cover jitter default-on, boundary values, and opt-out
- [x] Manual verification: `jitter: false` produces deterministic delays identical to prior behavior

### CODE REVIEW

File: packages/appkit/src/plugin/interceptors/retry.ts [hunk 1]
```diff
@@ -59,11 +61,13 @@ export class RetryInterceptor implements ExecutionInterceptor {
...
-    // exponential backoff
     const delay = this.initialDelay * 2 ** (attempt - 1);
+    const capped = Math.min(delay, this.maxDelay);
 
-    // max delay cap
-    return Math.min(delay, this.maxDelay);
+    if (!this.jitter) return capped;
+
+    // Full jitter: random value between 50% and 100% of the calculated delay
+    return capped * (0.5 + Math.random() * 0.5);
```
Comment (ditadi) [inline packages/appkit/src/plugin/interceptors/retry.ts]: For full jitter should be `capped * Math.random()` only

With the 0.5, you bound to half of the window.

File: packages/appkit/src/plugin/interceptors/retry.ts [hunk 2]
```diff
@@ -9,11 +9,13 @@ export class RetryInterceptor implements ExecutionInterceptor {
   private attempts: number;
   private initialDelay: number;
   private maxDelay: number;
+  private jitter: boolean;
```
Comment (ditadi) [inline packages/appkit/src/plugin/interceptors/retry.ts]: Do we need this flag? jitter is better in our cases for any production scenario

**Review by ditadi — APPROVED** [review]
Sounds good, I just added two points.

**Comment by MarioCadenas** [comment]
> Sounds good, I just added two points.

addressed, thank you!
